addon/components/high-charts.js

```diff
@@ -1,90 +1,105 @@
-import Component from '@ember/component';
+import Component from '@glimmer/component';
+
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
 import { getOwner } from '@ember/application';
-import { set, getProperties, get, computed } from '@ember/object';
 import { run } from '@ember/runloop';
+
+import buildOptions from '../utils/build-options';
 import { setDefaultHighChartOptions } from '../utils/option-loader';
 import { getSeriesMap, getSeriesChanges } from '../utils/chart-data';
-import layout from 'ember-highcharts/templates/components/high-charts';
-import merge from 'deepmerge';
 
-/* Map ember-highcharts modes to Highcharts methods
- * https://api.highcharts.com/class-reference/Highcharts.html
- */
-const CHART_TYPES = {
+const CHART_TYPES = Object.freeze({
   StockChart: 'stockChart',
   Map: 'mapChart',
-  undefined: 'chart'
-};
-
-export default Component.extend({
-  layout,
-  classNames: ['highcharts-wrapper'],
-  content: undefined,
-  mode: undefined,
-  chartOptions: undefined,
-  chart: null,
-  theme: undefined,
-  callback: undefined,
-
-  buildOptions: computed('chartOptions', 'content.[]', function() {
-    let theme = get(this, 'theme');
-    if (theme === undefined) {
-      theme = {};
-    }
+  undefined: 'chart',
+});
 
-    let passedChartOptions = get(this, 'chartOptions');
-    if (passedChartOptions === undefined) {
-      passedChartOptions = {};
-    }
+export default class HighCharts extends Component {
+  get content() {
+    return this.args.content ?? undefined;
+  }
 
-    let chartOptions = merge(theme, passedChartOptions);
-    let chartContent = get(this, 'content');
+  get chartOptions() {
+    return this.args.chartOptions ?? undefined;
+  }
 
-    // if 'no-data-to-display' module has been imported, keep empty series and leave it to highcharts to show no data label.
-    if (!get(this, 'content.length') && !Highcharts.Chart.prototype.showNoData) {
-      chartContent = [{
-        id: 'noData',
-        data: 0,
-        color: '#aaaaaa'
-      }];
-    }
+  get mode() {
+    return this.args.mode ?? undefined;
+  }
+
+  get theme() {
+    return this.args.theme ?? undefined;
+  }
+
+  get callback() {
+    return this.args.callback ?? undefined;
+  }
+
+  @tracked
+  el = undefined;
+
+  @tracked
+  chart = null;
+
+  get buildOptions() {
+    return buildOptions(this.theme, this.chartOptions, this.content);
+  }
+
+  drawAfterRender() {
+    run.scheduleOnce('afterRender', this, 'draw');
+  }
+
+  draw() {
+    const element = this.el?.querySelector('.chart-container');
 
-    let defaults = { series: chartContent };
+    // for any mode that is falsy ('', undefined, false), set it to default 'chart'
+    const mode = CHART_TYPES[this.mode] ?? CHART_TYPES.undefined;
+    const completeChartOptions = [this.buildOptions, this.callback];
 
-    return assign(defaults, chartOptions);
-  }),
+    if (element) {
+      // eslint-disable-next-line no-undef
```

Comment:
Instead of these lint overrides, why not leave `HighCharts` as a global in the eslintrc?

Reply:
updated!